integration/devmode/devmode_test.go

```diff
@@ -0,0 +1,291 @@
+			fmt.Sprintf("CORE_PEER_ADDRESS=%s", peerChaincodeAddress),
+			fmt.Sprintf("CORE_CHAINCODE_ID_NAME=%s", legacyChaincodeID),
+		}
+
+		cmd := exec.Command(chaincodeExecutePath, []string{"-peer.address", peerChaincodeAddress}...)
+		cmd.Env = append(cmd.Env, envs...)
+		chaincodeRunner = ginkgomon.New(ginkgomon.Config{
+			Name:              "chaincode",
+			Command:           cmd,
+			StartCheckTimeout: 15 * time.Second,
+		})
+		chaincodeProcess = ifrit.Invoke(chaincodeRunner)
+		Eventually(chaincodeProcess.Ready(), network.EventuallyTimeout).Should(BeClosed())
+
+		By("installing the chaincode")
+		nwo.InstallChaincodeLegacy(network, legacyChaincode, peers...)
+		By("instantiating the chaincode")
+		nwo.InstantiateChaincodeLegacy(network, channelName, orderer, legacyChaincode, org1peer0, peers...)
+		By("querying and invoking the chaincode")
+		RunQueryInvokeQuery(network, orderer, org1peer0, channelName, 100)
+
+		By("killing chaincode process")
+		chaincodeProcess.Signal(syscall.SIGKILL)
+		Eventually(chaincodeProcess.Wait(), network.EventuallyTimeout).Should(Receive())
+
+		By("querying chaincode after it is being killed")
+		sess, err := network.PeerUserSession(org1peer0, "User1", commands.ChaincodeQuery{
+			ChannelID: channelName,
+			Name:      "mycc",
+			Ctor:      `{"Args":["query","a"]}`,
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Eventually(sess, network.EventuallyTimeout).Should(gexec.Exit(0))
+		Expect(sess).To(gbytes.Say(strconv.Itoa(90)))
+
+		By("restarting the chaincode process")
+		cmd = exec.Command(chaincodeExecutePath, flags...)
+		cmd.Env = append(cmd.Env, envs...)
+		chaincodeRunner = ginkgomon.New(ginkgomon.Config{
+			Name:              "chaincode",
+			Command:           cmd,
+			StartCheckTimeout: 15 * time.Second,
+		})
+		chaincodeProcess = ifrit.Invoke(chaincodeRunner)
+		Eventually(chaincodeProcess.Ready(), network.EventuallyTimeout).Should(BeClosed())
+
+		By("querying and invoking the chaincode")
+		RunQueryInvokeQuery(network, orderer, org1peer0, channelName, 90)
+	})
+
+	It("executing chaincode in dev mode", func() {
+		chaincode = nwo.Chaincode{
+			Name:            "mycc",
+			Version:         "0.0",
+			Path:            components.Build("github.com/hyperledger/fabric/integration/chaincode/simple/cmd"),
+			Lang:            "binary",
+			PackageFile:     filepath.Join(testDir, "simplecc.tar.gz"),
+			Ctor:            `{"Args":["init","a","100","b","200"]}`,
+			SignaturePolicy: `OR ('Org1MSP.member')`,
+			Sequence:        "1",
+			InitRequired:    true,
+			Label:           "my_prebuilt_chaincode",
+		}
+
+		org1peer0 := network.Peer("Org1", "peer0")
+		peers := network.PeersWithChannel(channelName)
+		orderer := network.Orderer("orderer")
+
+		By("setting up the channel")
+		network.CreateAndJoinChannel(orderer, channelName)
+
+		By("enable new lifecycle capabilities")
+		nwo.EnableCapabilities(network, "testchannel", "Application", "V2_0", orderer, org1peer0)
+
+		By("packaging chaincode")
```

Comment:
We don't need to package the chaincode anymore since we aren't installing anything in this updated version of the test, right?

Reply:
it turns out `SetPackageIDFromPackageFile` in `ApproveChaincodeForMyOrg` is looking for packaged chaincode file to obtain chaincode PackagedID. 
changed the logic slightly, and fixed it.